### PR TITLE
RAC-237: add component generator

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

This PR add a new component generator and ensure that it stays up to date by testing it like other components

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: akeneo-design-system/bin/dsm [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  .help('h')
+  .alias('h', 'help')
+  .demandCommand(2).argv
+
+
+/**
+ * Copy Component files
+ */
+const copyComponentFolder = (source, target) => {
+  const DummyComponentName = source.split('/').pop();
```
Comment (ValentinMumble) [inline akeneo-design-system/bin/dsm:35]: Is there a reason for capitalized var name here?

File: akeneo-design-system/bin/dsm [hunk 2]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const yargs = require("yargs");
```
Comment (ValentinMumble) [inline akeneo-design-system/bin/dsm:4]:
```suggestion
const yargs = require('yargs');
```
Un ti coup de Prettier par là dessus

File: akeneo-design-system/generator/components/MyComponent/MyComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, {Ref} from 'react';
+import styled from 'styled-components';
+
+const MyComponentContainer = styled.span<{level: string}>``;
```
Comment (ValentinMumble) [inline akeneo-design-system/generator/components/MyComponent/MyComponent.tsx:4]:
```suggestion
//TODO be sure to select the appropriate container element here
const MyComponentContainer = styled.span<{level: string}>``;
```

**Review by ValentinMumble — APPROVED** [review]

File: akeneo-design-system/bin/dsm (hunk 1 above)
Comment (juliensnz) [reply akeneo-design-system/bin/dsm:35]: No, just to be consistant. I will remove it :)

File: akeneo-design-system/generator/components/MyComponent/MyComponent.unit.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import {MyComponent} from './MyComponent';
+import {render} from '../../storybook/test-util';
+import '@testing-library/jest-dom/extend-expect';
+
+test('it renders its children properly', () => {
```
Comment (SamirBoulil) [inline akeneo-design-system/generator/components/MyComponent/MyComponent.unit.tsx:6]: Did you ask your PM ?

**Review by SamirBoulil — APPROVED** [review]

**Pull request merged**